### ISSUE TITLE
[a11y] When focusing Widget created by showModalBottomSheet, focus is not immediately on contained actions

### ISSUE THREAD

**Issue by ChristianEdwardPadilla** [flutter/flutter#158474]
### Steps to reproduce

1. Enable voiceover on your platform.
2. Launch a modal bottom sheet via `showModalBottomSheet`.
3. Navigate to it via voiceover.
4. Observe that the dialog itself is focused instead of the contents of the dialog.

### Expected results

The interactive contents of the dialog should be focused first.

### Actual results

Instead of interactive contents being focused, the dialog itself is focused. This is at-odds with behavior seen on most apps (e.g. Google Home app).

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [showModalBottomSheet].
void main() => runApp(const BottomSheetApp());

class BottomSheetApp extends StatelessWidget {
  const BottomSheetApp({super.key});
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('Bottom Sheet Sample')),
        body: const BottomSheetExample(),
      ),
    );
  }
}

class BottomSheetExample extends StatelessWidget {
  const BottomSheetExample({super.key});
  @override
  Widget build(BuildContext context) {
    return Center(
      child: ElevatedButton(
        child: const Text('showModalBottomSheet'),
        onPressed: () {
          showModalBottomSheet<void>(
            context: context,
            builder: (BuildContext context) {
              return Container(
                height: 200,
                color: Colors.amber,
                child: Center(
                  child: Column(
                    mainAxisAlignment: MainAxisAlignment.center,
                    mainAxisSize: MainAxisSize.min,
                    children: <Widget>[
                      const Text('Modal BottomSheet'),
                      ElevatedButton(
                        child: const Text('Close BottomSheet'),
                        onPressed: () => Navigator.pop(context),
                      ),
                    ],
                  ),
                ),
              );
            },
          );
        },
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/0353183a-bf53-4353-9ccc-e6a52ef8a9a1

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Flutter (Channel google3, on macOS 14.7 
    • Framework revision 461ad0d3e0 (0 days ago), 2024-11-11T00:00:00.000
    • Engine revision d90e9f4718
    • Dart version ed9a5b1110
```

</details>

**Comment by darshankawar** [comment]
Thanks for the detailed report. Tried on Android and iOS using latest versions and was able to replicate the reported behavior. 

```
stable : 3.24.4
master : 3.27.0-1.0.pre.443
```

**Comment by chunhtai** [comment]
we have assigned sort key to modal barrier and modal scope to force the focus order. I think probably some weird merge logic that change the semantics node structure of modal barrier and modal scope that void the sort key.
